src/Libraries/Microsoft.Extensions.AI/ChatCompletion/ChatClientStructuredOutputExtensions.cs

```diff
@@ -176,4 +192,18 @@ public static async Task<ChatCompletion<T>> CompleteAsync<T>(
             }
         }
     }
+
+    private static bool SchemaRepresentsObject(JsonElement schemaElement)
+    {
+        foreach (var property in schemaElement.EnumerateObject())
+        {
+            if (property.NameEquals("type"u8))
```

Comment:
Do we want to support keywords containing arrays? `["object"]` is equivalent but it perhaps we can tolerate `["object","null"]`?

Reply:
Do you have a sense of when this situation would occur?

AFAIK even if we add a wrapper layer unnecessarily things will typically still work. So there's an argument that we should add the wrapper unless we can be confident it's not required. So we could say we don't have to detect all the cases where it's not required, and only deal with ones that will be particularly common.

Reply:
Given that we currently control schema generation, it won't ever occur for root-level schemas. Just posting the hypothetical question for completeness.

Reply:
OK, then I think it's fine already. There are many possible theoretical inputs we're not optimized for. Being optimized for the ones we think will actually happen should suffice :)